main.go

```diff
@@ -635,7 +637,20 @@ func newProxyDialContext(dialTimeout time.Duration) DialContext {
 // tlsClientSessionCacheSize is the cache size for TLS client sessions.
 const tlsClientSessionCacheSize = 100
 
-func clientTransport(ctx *cli.Context, enableTLS bool) http.RoundTripper {
+type roundTripperWrapper struct {
+	enableTLS  bool
+	cacert     string
+	clientCert string
+	clientKey  string
+	insecure   bool
+	rwLocker   sync.RWMutex
+	transport  http.RoundTripper
+}
+
+func (rtw *roundTripperWrapper) renew() {
+	rtw.rwLocker.Lock()
+	defer rtw.rwLocker.Unlock()
+	dnsCache.Refresh()
```

Comment:
If this does all the work, is it necessary to replace the transport. Can't the fix just be this line in `doHealthCheck()`, or am i missing something?

Reply:
Yes, but the dialer should pick that up, right?